tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,352 @@
+    {
+        script_run("virsh destroy $_");
+        script_run("virsh undefine $_");
+        script_run("virsh define $_.xml");
+	upload_logs "$_.xml";
+        script_run("rm -rf $_.xml");
+    }
+
+}
+
+sub restore_network {
+
+    my $network_mark  = "/etc/sysconfig/network/ifcfg-br0.new";
+    if (script_run("[[ -f $network_mark ]]") == 0){
+        assert_script_run("ip link set dev $virt_host_bridge down",60);
+        my $bridge_destroy = <<'EOF';
+#!/bin/bash
+ACTIVE_NET=$(ip a|awk -F': ' '/state UP/ {print $2}'|head -n1)
+interface="BOOTPROTO='dhcp'\nSTARTMODE='auto'"
+echo -e $interface >/etc/sysconfig/network/ifcfg-$ACTIVE_NET
+cat /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+rm -rf /etc/sysconfig/network/ifcfg-br0*
+systemctl restart network.service;
+EOF
+         script_output($bridge_destroy, 900);
+    }
+
+}
+
+sub destroy_vir_network {
+
+    my $wait_script       = "30";
+    my $get_vnet_name     = "virsh net-list --all| tail -2| head -1| awk \'{print \$1}\'";
+    my $vnet_name         = script_output($get_vnet_name, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my @vnet_name_array = split(/\n+/, $vnet_name);
+    foreach (@vnet_name_array) {script_run("virsh net-destroy $_"); script_run("rm -rf $_.xml");}
+
+    my $default_path  = "/root/libvirt_default.xml";
+    if (script_run("[[ -f $default_path ]]") == 0){assert_script_run("virsh net-define $default_path",60);assert_script_run("rm -rf $default_path");}
+
+}
+
+sub run {
+
+    #Prepare VM HOST SERVER Network Interface Configuration
+    #for libvirt virtual network testing 
+    prepare_network;
+
+    #Debug br123
+    restore_standalone;
+
+    #Remove SSH old files and Generate the key pair
+    assert_script_run "rm -rf ~/.ssh/* || true;ssh-keygen -t rsa -P '' -f ~/.ssh/id_rsa";
+ 
+    #Install required packages
+    zypper_call '-t in iptables iputils bind-utils sshpass';
+
+    #Guest SSH Setup
+    my $gi_guest = '';
+    foreach my $guest (keys %xen::guests) {
+        #Archive deployed Guests
+        assert_script_run("virsh dumpxml $guest > $guest.xml");
+        record_info "$guest", "Establishing SSH connection to $guest";
+        assert_script_run("virsh start $guest", 60);
+        save_screenshot;
+        $gi_guest = script_output("sleep 40;journalctl | grep dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+	#Copy the VM host SSH Key to guest systems
+        assert_script_run("sshpass -p novell ssh-copy-id -f root\@$gi_guest", 60);
+	#Prepare the new guest network interface files for libvirt virtual netwok
+        assert_script_run("ssh root\@$gi_guest 'cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-eth1;cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-eth2'");
+
+     }
+
+##BRIDGE - USING AN EXISTING BRIDGE ON VM HOST SERVER
+##Prepare Test
+my $vnet_host_bridge = <<'EOF';
+<network>
+<name>vnet_host_bridge</name>
+<forward mode="bridge"/>
+<bridge name="BRI"/>
```

Comment:
Did you use `BRI` instead of something like `br1` by purpose? It is probably good idea

Reply:
It does depend that 'br0' as the default host bridge on VM Server. Here "BRI" just only 
variable for 'br0'.

Reply:
OK, this is OK.